mechanics/RELEASE-VERSIONING-PRINCIPLES.md

```diff
@@ -0,0 +1,953 @@
+  default.
+- Removed means support no longer exists within the open source release
+
+**Note**: subject to change based on verification of (knative, kube) version
+tuple
+
+<table>
+  <tr>
+   <td><strong>Knative Version</strong>
+   </td>
+   <td><strong>Release Date</strong>
+   </td>
+   <td><strong>End of LIfe Date</strong>
+   </td>
+   <td><strong>Min K8s Version</strong>
+   </td>
+   <td><strong>Storage Type</strong>
+   </td>
+   <td><strong>Force Upgrade</strong>
+   </td>
+   <td><strong>API Endpoints</strong>
+   </td>
+   <td><strong>API Type Served</strong>
+   </td>
+   <td><strong>Availability</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.17.x
+   </td>
+   <td rowspan="3" >2020-08-18
+   </td>
+   <td rowspan="3" >2021-02-02
+   </td>
+   <td rowspan="3" >1.18.x
+   </td>
+   <td rowspan="3" >v1
+   </td>
+   <td rowspan="3" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td><strong>removed</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>Lemonade
+   </td>
+   <td><strong>removed</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.16.x
+   </td>
+   <td rowspan="3" >2020-07-07
+   </td>
+   <td rowspan="3" >2020-12-22
+   </td>
+   <td rowspan="3" >1.18.x
+   </td>
+   <td rowspan="3" >v1
+   </td>
+   <td rowspan="3" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>Lemonade
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.15.x
+   </td>
+   <td rowspan="3" >2020-05-26
+   </td>
+   <td rowspan="3" >2020-11-10
+   </td>
+   <td rowspan="3" >1.16.x
+   </td>
+   <td rowspan="3" >v1
+   </td>
+   <td rowspan="3" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>Lemonade
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.14.x
+   </td>
+   <td rowspan="3" >2020-04-14
+   </td>
+   <td rowspan="3" >2020-09-29
+   </td>
+   <td rowspan="3" >1.16.x
+   </td>
+   <td rowspan="3" >v1
+   </td>
+   <td rowspan="3" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>Lemonade
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.13.x
+   </td>
+   <td rowspan="3" >2020-03-03
+   </td>
+   <td rowspan="3" >2020-08-18
+   </td>
+   <td rowspan="3" >1.15.x
+   </td>
+   <td rowspan="3" >v1
```

Comment:
storage type for 0.12.x is actually v1alpha1

Reply:
Done.